From 1cc2a6ddcdc4ef9b322c9ee29410d0248e0bea35 Mon Sep 17 00:00:00 2001
From: Stefan Reisinger <dev@stefan-reisinger.at>
Date: Tue, 15 Sep 2026 12:25:29 +0200
Subject: [PATCH] board: mediatek: bpi-r2: add FIT selectors

Add separate FIT configuration selectors for production and recovery.
This lets production enable a device tree overlay while recovery keeps
using the base configuration and remains compatible with older images.

Signed-off-by: Stefan Reisinger <dev@stefan-reisinger.at>
---
 defenvs/bananapi_bpi-r2_env | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/defenvs/bananapi_bpi-r2_env b/defenvs/bananapi_bpi-r2_env
index b445970..813e53f 100644
--- a/defenvs/bananapi_bpi-r2_env
+++ b/defenvs/bananapi_bpi-r2_env
@@ -2,6 +2,9 @@ ipaddr=192.168.1.1
 serverip=192.168.1.254
 loadaddr=0x88000000
 dtaddr=0x83f00000
+bootconf=config-1
+bootconf_ds3231=config-1#mt7623n-bananapi-bpi-r2-ds3231
+bootconf_recovery=config-1
 console=earlycon=uart8250,mmio32,0x11004000 console=ttyS2,115200 console=tty1
 initrd_high=0xafffffff
 part_default=3
@@ -22,9 +24,9 @@ bootmenu_1=Boot system via TFTP.=run boot_tftp ; run bootmenu_confirm_return
 boot_default=if env exists flag_recover ; then else run bootcmd ; fi ; run boot_recovery ; setenv replacevol 1 ; run boot_tftp_forever
 boot_first=if button factory ; then led $bootled_rec on ; run boot_tftp_recovery ; setenv flag_recover 1 ; run boot_default ; fi ; bootmenu
 boot_tftp_forever=led bpi-r2:pio:blue on ; while true ; do run boot_tftp_recovery ; sleep 1 ; done
-boot_tftp_production=tftpboot $loadaddr $bootfile_upg && env exists replacevol && iminfo $loadaddr && run mmc_write_production ; if env exists noboot ; then else bootm $loadaddr ; fi
-boot_tftp_recovery=tftpboot $loadaddr $bootfile && env exists replacevol && iminfo $loadaddr && run mmc_write_recovery ; if env exists noboot ; then else bootm $loadaddr ; fi
-boot_tftp=tftpboot $loadaddr $bootfile && bootm $loadaddr
+boot_tftp_production=tftpboot $loadaddr $bootfile_upg && env exists replacevol && iminfo $loadaddr && run mmc_write_production ; if env exists noboot ; then else bootm $loadaddr#$bootconf ; fi
+boot_tftp_recovery=tftpboot $loadaddr $bootfile && env exists replacevol && iminfo $loadaddr && run mmc_write_recovery ; if env exists noboot ; then else bootm $loadaddr#$bootconf_recovery ; fi
+boot_tftp=tftpboot $loadaddr $bootfile && bootm $loadaddr#$bootconf_recovery
 boot_mmc=run boot_production ; run boot_recovery
 emmc_init=run emmc_init_bl && run emmc_init_openwrt ; env default bootcmd ; saveenv
 emmc_init_bl=run sdmmc_read_emmc_hdr && run emmc_write_hdr && run sdmmc_read_preloader && run emmc_write_preloader && run sdmmc_read_uboot && run emmc_write_uboot
@@ -51,11 +53,11 @@ sdmmc_write_recovery=iminfo $fileaddr && mmc dev 1 && part start mmc 1 $part_rec
 _checkbootedfrom=setenv _checkbootedfrom ; if itest.l *81dffff0 == 434d4d65 ; then setenv bootedfrom eMMC ; else setenv bootedfrom SD ; fi
 _init_env=setenv _init_env ; setenv _create_env ; saveenv ; saveenv
 _firstboot=setenv _firstboot ; led $bootled_pwr off ;led $bootled_rec on ; run _checkbootedfrom _switch_to_menu _update_bootcmd _update_bootcmd2 _init_env boot_first
-_set_bootcmd_sdmmc=setenv boot_production "led $bootled_rec off ; led $bootled_pwr on ; run sdmmc_read_production && bootm $loadaddr ; led $bootled_pwr off"
-_set_bootcmd_emmc=setenv boot_production "led $bootled_rec off ; led $bootled_pwr on ; run emmc_read_production && bootm $loadaddr ; led $bootled_pwr off"
+_set_bootcmd_sdmmc=setenv boot_production 'led $bootled_rec off ; led $bootled_pwr on ; run sdmmc_read_production && bootm $loadaddr#$bootconf ; led $bootled_pwr off'
+_set_bootcmd_emmc=setenv boot_production 'led $bootled_rec off ; led $bootled_pwr on ; run emmc_read_production && bootm $loadaddr#$bootconf ; led $bootled_pwr off'
 _update_bootcmd=setenv _update_bootcmd ; if test "$bootedfrom" = "SD" ; then run _set_bootcmd_sdmmc ; else run _set_bootcmd_emmc ; fi ; setenv _set_bootcmd_sdmmc ; setenv _set_bootcmd_emmc
-_set_bootcmd2_sdmmc=setenv boot_recovery "led $bootled_pwr off ; led $bootled_rec on ; run sdmmc_read_recovery && bootm $loadaddr ; led $bootled_rec off"
-_set_bootcmd2_emmc=setenv boot_recovery "led $bootled_pwr off ; led $bootled_rec on ; run emmc_read_recovery && bootm $loadaddr ; led $bootled_rec off"
+_set_bootcmd2_sdmmc=setenv boot_recovery 'led $bootled_pwr off ; led $bootled_rec on ; run sdmmc_read_recovery && bootm $loadaddr#$bootconf_recovery ; led $bootled_rec off'
+_set_bootcmd2_emmc=setenv boot_recovery 'led $bootled_pwr off ; led $bootled_rec on ; run emmc_read_recovery && bootm $loadaddr#$bootconf_recovery ; led $bootled_rec off'
 _update_bootcmd2=setenv _update_bootcmd2 ; if test "$bootedfrom" = "SD" ; then run _set_bootcmd2_sdmmc ; else run _set_bootcmd2_emmc ; fi ; setenv _set_bootcmd2_sdmmc ; setenv _set_bootcmd2_emmc
 _switch_to_menu=setenv _switch_to_menu ; setenv bootdelay 3 ; setenv bootmenu_delay 3 ; setenv bootmenu_0 $bootmenu_0d ; setenv bootmenu_0d ; run _bootmenu_update_title
 _bootmenu_update_title=setenv _bootmenu_update_title ; setenv bootmenu_title "$bootmenu_title  [0;36m[$bootedfrom][0m    [33m$ver[0m" ; run _set_bm2
